app/validation/validators.py

```diff
@@ -296,29 +296,26 @@ def _is_first_relative_delta_largest(relativedelta1, relativedelta2):
     def _build_range_length_error(self, period_object):
         error_message = ''
         if 'years' in period_object:
-            error_message = self.return_error_component(error_message, period_object['years'], 'year')
+            unit = ngettext('year', 'years', period_object['years'])
+            error_message = self.return_error_component(error_message, period_object['years'], unit)
         if 'months' in period_object:
-            error_message = self.return_error_component(error_message, period_object['months'], 'month')
+            unit = ngettext('month', 'months', period_object['months'])
+            error_message = self.return_error_component(error_message, period_object['months'], unit)
         if 'days' in period_object:
-            error_message = self.return_error_component(error_message, period_object['days'], 'day')
+            unit = ngettext('day', 'days', period_object['days'])
+            error_message = self.return_error_component(error_message, period_object['days'], unit)
 
         return error_message
 
     @staticmethod
     def return_error_component(error_message, number, unit):
-        unit_plurals = {
-            'year': [_('years'), _('year')],
-            'month': [_('months'), _('month')],
-            'day': [_('days'), _('day')]
-        }
-        plural = unit_plurals[unit][(1 if number == 1 else 0)]
 
         if error_message != '':
             error_message = error_message + ', '
 
-        error_message = error_message + str(number) + ' ' + _(plural)
+        error_message = error_message + str(number) + ' ' + unit
```

Comment:
I'd suggest including the count number with the translation usually instead of concatenating it.

error_message += ngettext('%(num)s month', '%(num)s months', number)

This approach lets languages do the equivalent of saying "a year" instead of "1 year".

Just something to consider if it can be done without compromising code readability?